gdal/ogr/ogrsf_frmts/lvbag/ogrlvbaglayer.cpp

```diff
@@ -505,17 +535,70 @@ void OGRLVBAGLayer::EndElementCbk( const char *pszName )
                 reinterpret_cast<OGRGeometry *>(OGR_G_CreateFromGML(osElementString.c_str())) };
             if( poGeom && !poGeom->IsEmpty() )
             {
-                if( !bSchemaOnly )
+                // The specification only accounts for 2-dimensional datasets
+                if( poGeom->Is3D() )
+                    poGeom->flattenTo2D();
+
+#if defined(HAVE_SFCGAL) || defined(HAVE_GEOS)
```

Comment:
MakeValid() is only implemented for GEOS >= 3.8, so testing for HAVE_SFCGAL will not bring anything

Reply:
I'll check GEOS version. Also this test fails on Windows and I don't know why. Is it because of this version?

Reply:
> Also this test fails on Windows and I don't know why. Is it because of this version?

Possibly. I can see "ERROR 6: GEOS 3.8 or later needed for MakeValid.". Actually none of the CI target must have GEOS 3.8 yet

Reply:
I've listed `$(GEOS_CFLAGS)` in the makefile but `GEOS_VERSION_MAJOR` never gets passed down. Are these defines only set in later versions?

Reply:
you need to include ogr_geos.h

Reply:
Skipped the test for now